src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -284,6 +284,14 @@ memory::ptr primitive_inst::allocate_output(engine& _engine, memory_pool& pool,
     if (total_device_input_mem_size > _engine.get_device_info().max_global_mem_size)
         usm_device_allocatable = false;
 
+    bool memory_reuse_by_user = true;
+    for (auto user : _node.get_users()) {
+        if (user->get_selected_impl()->can_reuse_memory == false) {
+            memory_reuse_by_user = false;
+        }
+    }
```

Comment:
I think you may check the "optimized" layers b/w current node and the users. 
For example, 
cur_node -> userA (optimized reshape) -> userB
In this case the userB should be checked. 
Also, userA might not have selected impl. Didnt you need to check whether it has impl or not?

Reply:
Added conditions when user->get_selected_impl()=nullptr. Thanks!